node_modules/gh-orgunit/lib/internal/dao.js

```diff
@@ -760,3 +763,402 @@ var _exportOrgUnitJSON = function(rootId, orgUnits, callback) {
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error object, if any
+ * @api private
+ */
+var _importOrgUnit = function(orgUnit, parent, callback) {
+    var parentId = null;
+    if (parent) {
+        parentId = parent.id;
+        orgUnit.AppId = parent.AppId;
+    }
+
+    _updateOrCreateOrgUnit(orgUnit, parent, function(err, createdOrgUnit) {
+        if (err) {
+            log().error({'err': err, 'externalId': orgUnit.externalId}, 'Failed to import organisational unit');
+            return callback(err);
+        }
+
+        // Import the organisational unit's child organisational units (if any)
+        _importOrgUnitChildren(orgUnit.children, createdOrgUnit, function(err) {
+            if (err) {
+                return callback(err);
+            }
+
+            // Import the series under this organisational unit
+            _importOrgUnitSeries(orgUnit.series, createdOrgUnit, function(err) {
+                if (err) {
+                    return callback(err);
+                }
+
+                // TODO: Import events directly associated to organisational units
+                return callback();
+            });
+        });
+    });
+};
+
+/**
+ * Update or create an organisational unit
+ *
+ * @param  {Object}     orgUnitToImport             The organisational unit to import
+ * @param  {Object}     parent                      The paren organisational unit to which the organisational unit belongs
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error object, if any
+ * @param  {Event}      callback.persistedOrgUnit   The persisted organisational unit
+ * @api private
+ */
+var _updateOrCreateOrgUnit = function(orgUnitToImport, parent, callback) {
+    if (orgUnitToImport.id) {
+        // Update the organisational unit
+        getOrgUnit(orgUnitToImport.id, false, function(err, persistedOrgUnit) {
+            if (err) {
+                return callback(err);
+            }
+
+            var update = {
+                'displayName': orgUnitToImport.displayName,
+                'description': orgUnitToImport.description,
+                'type': orgUnitToImport.type,
+                'metadata': orgUnitToImport.metadata,
+                'published': orgUnitToImport.published
+            };
+            return updateOrgUnit(persistedOrgUnit, update, callback);
+        });
+    } else {
+        // Create the organisational unit
+        var opts = {
+            'description': orgUnitToImport.description,
+            'externalId': orgUnitToImport.externalId,
+            'metadata': orgUnitToImport.metadata,
+            'published': orgUnitToImport.published,
+            'parentId': parent.id
+        };
+        return createOrgUnit(orgUnitToImport.AppId, orgUnitToImport.displayName, orgUnitToImport.type, parent.GroupId, opts, callback);
```

Comment:
FWIW, I think the fact that we look at an AppId in the data rather than hand it into the DAO as a DB object tripped me up on the whole validation thing. Preferrably we could hand the `app` as the first parameter of the import method(s), but if we need to support multi-app imports then maybe that isn't feasible.